packages/dds/tree/src/shared-tree/treeApi.ts

```diff
@@ -66,6 +66,20 @@ export interface TreeApi extends TreeNodeApi {
 		tree: TView,
 		transaction: (root: TView["root"]) => void | "rollback",
 	): void;
+
+	/**
+	 * Check if the subtree defined by `parent`	contains `child`.
+	 *
+	 * @returns true if and only if the subtree rooted at `parent` contains `child`.
+	 * @remarks
+	 * Parent is considered to contain itself, so the case where `parent === child` returns true.
+	 *
+	 * This is handy when checking if moving parent into child would create a cycle and thus is invalid.
+	 *
+	 * This check walks the parents of child looking for parent,
+	 * and thus runs in time proportional to the depth of child in the tree.
+	 */
+	contains(parent: TreeNode, child: TreeNode): boolean;
```

Comment:
I find `contains` to be very readable and better than the alternatives you mentioned. I think we should also consider `has`, since that has precedent in JS maps. However, I would still vote for `contains`, which has precedence in the DOM tree - and that's a more analogous comparison. Note that the [DOM doc](https://developer.mozilla.org/en-US/docs/Web/API/Node/contains) for `contains` also talks in terms of "descendants":

"Returns true if other is an inclusive descendant of node, and false otherwise."

Reply:
given the risk of someone thinking this is shallow and not deep, contains seems better than has.

I didn't know about that DOM API: aligning with its terminology around "inclusive descendant" and the parameter names it uses sounds like a great idea.

Reply:
I have Adjusted the names and docs inspired by that DOM API and your comment.